plugin/resurrect/pane_tree.lua

```diff
@@ -94,16 +97,33 @@ local function insert_panes(root, panes)
 		return nil
 	end
 
-	root.cwd = root.pane:get_current_working_dir().file_path
-	if is_windows then
-		root.cwd = root.cwd:gsub("^/([a-zA-Z]):", "%1:")
-	end
-	root.process = root.pane:get_foreground_process_name()
-	if pub.get_shell_process(root.process) then
-		root.text = root.pane:get_lines_as_escapes(root.pane:get_dimensions().scrollback_rows)
+	local domain = root.pane:get_domain_name()
+	if not string.sub(domain, 1, 3) == "SSH" or not domain == "local" then
+		wezterm.log_warn("Unknown domain: " .. domain)
```

Comment:
Just out of curiosity, can there be other domains than SSH or local? Could it be TSL or SERIAL?

Reply:
I hadn't thought aboutother domains, but you are absolutely right that there could be TLS, Serial, WSL, maybe others.

Perhaps it would be better here to use [domain:is_spawnable](https://wezfurlong.org/wezterm/config/lua/MuxDomain/is_spawnable.html) here for the check, and only save the domain key/value if it's spawnable.

Then within `window_state.lua` only call the `spawn_tab` function if there is a domain or non-empty process. I can `wezterm.log_warn()` and emit `resurrect.error` if the domain isn't spawnable.

I don't have a way to test TLS, serial or WSL so hopefully others can help out. Theoretically this should work.